posts/basic-probabliity-in-lean.md

```diff
@@ -0,0 +1,252 @@
+
+To express independence of sets, use `IndepSet` (two sets) and `iIndepSet` (family of sets). For sigma-algebras, `Indep` and `iIndep`.
+
+#### Conditional independence
+
+The way to write that two random variables `X : Ω → E` and `Y : Ω → F` are conditionally independent given a sub-sigma-algebra `m` with respect to the measure `P` is `CondIndepFun m hm X Y P`, in which `hm : m ≤ mΩ` is a proof that `m` is a sub-sigma-algebra of the sigma-algebra `mΩ` of `Ω`.
+We would omit the measure argument in a `MeasureSpace`.
+That definition requires that `Ω` be standard Borel space.
+
+To write that `X : Ω → E` and `Y : Ω → F` are conditionally independent given a third random variable `Z : Ω → G` (with measurability assumption `hZ : Measurable Z`), write that they are independent given the sigma-algebra generated by `Z`.
+That is, for `mG : MeasurableSpace G` the sigma-algebra on `G`, write `CondIndepFun (mG.comap Z) hZ.comap_le X Y P`.
+As of writing this blog post, there is no shorter spelling of that fact.
+
+For families of functions, use `iCondIndepFun`.
+For sets, use `CondIndepSet` (two sets) and `iCondIndepSet` (family of sets). For sigma-algebras, `CondIndep` and `iCondIndep`.
+
+### Stochastic processes, filtrations, martingales
+
+A stochastic process with real values (for example) is a function `X : ι → Ω → ℝ` from an index set `ι` to random variables `Ω → ℝ`.
+
+A filtration on the index set can be defined with `ℱ : Filtration ι m`, in which `m` is a sigma-algebra on `Ω`. `ℱ` is a monotone sequence of sub-sigma-algebras of `m`. The sigma-algebra at index `i` is `ℱ i`.
+
+We can state that a process `X` is adapted to the filtration `ℱ` with `adapted ℱ X`.
+We can write that it is a martingale with respect to `ℱ` and the measure `P : Measure Ω` with `Martingale X ℱ P`.
+
+A stopping time with respect to a filtration `ℱ` is a function `τ : Ω → ι` such that for all `i`, the preimage of `{j | j ≤ i}` along `τ` is measurable with respect to `ℱ i`. 
+`IsStoppingTime ℱ τ` states that `τ` is a stopping time.
+
+Remark: there is an issue with the current definition of stopping times, which is that if the index set of the filtration `ι` is the set of natural numbers `ℕ` then `τ` has to take values in `ℕ`. In particular it cannot be infinite.
+That can cause friction with informal stopping times which are often understood to be infinite when they are not well defined.
+
+### Transition kernels
+
+A transition kernel `κ` from a measurable space `E` to a measurable space `F` is a measurable function from `E` to `Measure F`. That is, for every measurable set `s` of `F`, the function `fun x ↦ κ x s` is measurable.
+Transition kernels are represented in Mathlib by the type `Kernel E F`.
+A kernel such that all measures in its image are probability measures is called a Markov kernel. This is denoted by `IsMarkovKernel κ`.
+There are other typeclasses for kernels that are finite (`IsFiniteKernel`) or s-finite (countable sum of finite, `IsSFiniteKernel`).
```

Comment:
None of these names are fully qualified. What about either fully qualifying the names or turning the code snippets into links to the docs?